types/d3-shape/index.d.ts

```diff
@@ -1865,14 +1865,371 @@ export const curveStepAfter: CurveFactory;
+
+    /**
+     * Returns the current target node accessor function.
+     * The default target accessor function returns a two element array [x, y].
+     */
+    target(): (this: This, d: LinkDatum, ...args: any[]) => NodeDatum;
+    /**
+     * Sets the target accessor to the specified function and returns this link generator.
+     *
+     * @param target Target node accessor function. The accessor function is invoked in the same "this" context as the generator was invoked in and
+     * receives the same arguments that were passed into the link generator. The default target accessor function returns a two element array [x, y].
+     */
+    target(target: (this: This, d: LinkDatum, ...args: any[]) => NodeDatum): this;
+
+    /**
+     * Returns the current x-accessor, which defaults to a function accepting an number array
+     * as its argument an returning the first element of the array.
+     */
+    x(): (this: This, node: NodeDatum, ...args: any[]) => number;
+    /**
+     * Sets the x-accessor to the specified function and returns this link generator.
+     *
+     * @param x x-coordinate accessor function. The accessor function is invoked in the same "this" context as the generator was invoked in and
+     * receives as its first argument a node object followed by all additional arguments that were passed into the link generator.
+     */
+    x(x: (this: This, node: NodeDatum, ...args: any[]) => number): this;
+
+    /**
+     * Returns the current y-accessor, which defaults to a function accepting an number array
+     * as its argument an returning the second element of the array.
+     */
+    y(): (this: This, node: NodeDatum, ...args: any[]) => number;
+    /**
+     * Sets the y-accessor to the specified function and returns this link generator.
+     *
+     * @param y y-coordinate accessor function. The accessor function is invoked in the same "this" context as the generator was invoked in and
+     * receives as its first argument a node object followed by all additional arguments that were passed into the link generator.
+     */
+    y(y: (this: This, node: NodeDatum, ...args: any[]) => number): this;
+
+    /**
+     * Returns the current rendering context, which defaults to null.
+     */
+    context(): CanvasRenderingContext2D | null;
+    /**
+     * Sets the rendering context and returns this link generator.
+     *
+     * If the context is not null, then the generated link is rendered to this context as a sequence of path method calls.
+     *
+     * @param context The rendering context.
+     */
+    context(context: CanvasRenderingContext2D): this;
+    /**
+     * Sets the rendering context to null and returns this link generator.
+     *
+     * A path data string representing the generated link will be returned when the generator is invoked with data.
+     *
+     * @param context null, to remove rendering context.
+     */
+    context(context: null): this;
```

Comment:
I don't consider this a blocker for now, but it's something to consider: we should really clean these up to use unions, since you can't preform a round trip:

```ts
const context = link.context();

// fails because the overloads aren't compatible with the return value of `.context()`:
link.context(context);
```

I'd like to see a cleanup so that we don't have to keep disabling `unified-signatures` in tslint.json.